### ISSUE TITLE
Proposal: Copy settings of running container and bake it to image with commit flag

### ISSUE THREAD

**Issue by jas99** [docker/docker#9917]
In addition to https://github.com/docker/docker/issues/3285#issuecomment-68619672
Like to have a flag while committing with ability to copy settings such as port opened, volumes mounted etc of running container and bake it to chain image.

Thx
Jaspreet

**Comment by jas99** [comment]
@cpuguy83 Never said all this info is stored or can be stored in image file; but while container is running it knows the info and the child image should be allowed to copy these setting just as Dockerfile allow to set up default settings in image.

**Comment by cpuguy83** [comment]
What is a "child" in this context?

**Comment by jas99** [comment]
Like Parent image ---(RUN with cli flags for settings ports etc)---> Container (edit setting in running container https://github.com/docker/docker/issues/3285#issuecomment-68619672 hopefully in future  ) --(Commit with copy settings flag)--> Child image(with default port etc as from running container; with similar effect if it would have been created with Docker file)

**Comment by cpuguy83** [comment]
Yes, this is exactly what I'm saying cannot be done.
Docker very specifically does not commit these settings, as they are host specific settings, and can cause runtime conflicts as well.

**Comment by jas99** [comment]
Cool; Since I just have high level idea (you know like project have feature and designs derived from git chroot and solaris zones or lxc's ); your points should be justified. They must be work on project :)

Although I am still little confused that Dockerfile can build preconfigured image but you can't inherent settings while committing.

I am pretty sure am missing something but still want this feature :)

**Comment by baptistedonaux** [comment]
+1

**Comment by cpuguy83** [comment]
These are runtime options that you are talking about, they cannot be stored in the image, and this is on purpose.

**Comment by jas99** [comment]
@cpuguy83 Hey I have question for you; if you don't mind.

Images created by Dockerfile has default settings. Is this metadata is hashed into the image?

**Comment by cpuguy83** [comment]
@jas99 `docker inspect <image>`

Please keep in mind this is not a support channel and every maintainer gets an email for each and every comment.

For support questions, #docker on IRC or the docker-user mailing list are both great resources.

**Comment by jas99** [comment]
@cpuguy83 
Already on IRC :)
I know docker inspect; am pretty comfortable with rest of commands and kinna medium at skill from usage point of view.

Okay lets say with this issue https://github.com/docker/docker/issues/3285#issuecomment-68619672 pending we are only dealing with inheritance stuff and I don't talk about it in my purposal.

Let me rephrase my purposal.

Command 
docker commit --inherit(-i) "container" 

Effect
> commit find parent image of running container
> commit inspects parent image
> commit hashes some of that meta data to new image

Can you please rephare where technical difficulty lies?

Jas

**Comment by cpuguy83** [comment]
@jas99 This is exactly what `docker commit` does today.

**Comment by jas99** [comment]
@cpuguy83 Sorry; don't use Dockerfile that often; yet  :)
sorry about your troubles; you can scrap this proposal.

Jas

**Comment by jas99** [comment]
thx
